### ISSUE TITLE
Incorrect actor resolving through middleman

### ISSUE THREAD

**Issue by Brewir** [actor-framework/actor-framework#1065]
I found some cases where the header is not correctly resolved when actor are exchanging through network.
```
consume scheduled_actor.cpp:681 RECEIVE ; FROM = null:pointer ; STAGES = [] ; CONTENT = (new_data_msg(229,...
```
with header
```
handle instance.cpp:82 hdr = header(direct_message, 0, 208, 1152921504606847285, 9687, 125)
```
is then decoded as
```
enqueue scheduled_actor.cpp:157 SEND ; TO = 125@C932DC656159D53CD8263B48164DFD76C00FD100#25841 ; FROM = null:pointer ; STAGES = [] ; CON        TENT = ('pxy_d_msg', ...
```

**Comment by Brewir** [comment]
Thanks a lot for this quick reaction

**Comment by Neverlord** [comment]
Let's not cheer too early and wait if the PR actually fixes the bug for you. 🙂

**Comment by Neverlord** [comment]
Fixed via #1070.
